src/main/io/gps_ublox.c

```diff
@@ -392,8 +392,9 @@ static void configureGNSS10(void)
 
             // Beidou
             {UBLOX_CFG_SIGNAL_BDS_ENA, gpsState.gpsConfig->ubloxUseBeidou},
-            {UBLOX_CFG_SIGNAL_BDS_B1_ENA, gpsState.gpsConfig->ubloxUseBeidou},
-            {UBLOX_CFG_SIGNAL_BDS_B1C_ENA, 0},
+            // Use B1 for Beidou without Glonass; B1C needed for concurrent use of Beidou and Glonass
+            {UBLOX_CFG_SIGNAL_BDS_B1_ENA,  gpsState.gpsConfig->ubloxUseBeidou &&  ! gpsState.gpsConfig->ubloxUseGlonass},
+            {UBLOX_CFG_SIGNAL_BDS_B1C_ENA, gpsState.gpsConfig->ubloxUseBeidou && gpsState.gpsConfig->ubloxUseGlonass},
```

Comment:
What about my m9 that supports both? maybe add a check for hwversion != 10?

Reply:
The current check is:

if(gpsState.hwVersion >= UBX_HW_VERSION_UBLOX10 || (gpsState.swVersionMajor>=23 && gpsState.swVersionMinor >= 1)) {
            configureGNSS10()

Which does seem a bit odd to me - checking whether it's the cut down hardware OR has current firmware. Anyone know why it's currently doing that?

Reply:
B1 works with  M9
Possibly even both B1 and B1C at the same time.

Maybe separating the Beidou options in BeidouB1 and BeidouB1C would be better, but that would need to happen in INAV 8.0